proposals/Reliable Mutability Primitive.md

```diff
@@ -0,0 +1,168 @@
+
+##### Impact
+_How directly important is the outcome to our top-level mission?_
+
+
+🔥🔥🔥 This would enable building competitive software that does not introduce custom server components (Standard [Pinning Services API][] provides data persistance and standard mutability primitive provides updating mechanism).
+
+It would also give an edge to the whole ecosystem, because [there is no step three](https://www.youtube.com/watch?v=YHzM4avGrKI) (as in no servers to run or maintain).
+
+
+<!--
+Explain why you have chosen this rating
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+##### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+
+<!-- Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+🎯🎯 - This would enable projects that need to have a canonical identifier for evolving dataset. It would also provide a greater interop across products built around IPFS by removing need for custom incompatibile solutions that teams need to implemente today.
+
+
+##### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+3 - As high confidence as we can have without doing actual user studies. We know teams that have tried IPNS but found it unrelaible, ultimately rolling out an alternative solution. We also know teams that have evaluated IPFS but chose alternative due to lack of reliable mutability story.
+
+
+
+### Project definition
+##### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Implement third party IPNS republishing (see [ipfs/go-ipfs#4435](https://github.com/ipfs/go-ipfs/issues/4435))
+- Collaborate with Pinning Services to make this part of the API.
+- Fix IPNS publishing in browsers.
+- Enable pubsub by default + IPNS over pubsub.
+- Implement & deploy strategic republishing system on PL hosted nodes that will republish popular content and drop stalled one. So that IPNS works even thought pinning services.
+- (Nice to have) Implement and rollout delegated publishing to enable updating from multiple devices.
+- (Nice to have) Implement and rollout versioning system so that nodes can see full update history.
+
+##### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+Publishing IPNS record takes just a few miliseconds. Shutting laptop and loading `https://ipfs.io/ipns/{key}` week later on other device loads the content published as fast as loading corresponding `https://ipfs.io/ipfs/{cid}`.
```

Comment:
I'd like to apply below change, to make it extra clear the goal here is to solve IPNS rot: 

```suggestion
- Publishing IPNS record takes just a few miliseconds. 
- Shutting laptop and loading `https://ipfs.io/ipns/{key}` week later on other device loads the content published as fast as loading corresponding `https://ipfs.io/ipfs/{cid}`.
- It is possible to pin IPNS address to keep it around and ensure it resolves even when original publisher is gone (AKA solving IPNS rot).
  - Node follows IPNS record updates and keeps re-publishing latest one if original author disappears.
  - Node follows content updates behind the IPNS record, and update content pin every time IPNS record changes. 
  - It is possible to opt-out from this behavior to re-publish IPNS record for `en.wikipedia-on-ipfs.org` without pinning all the data.
```